src/posthog-surveys.ts

```diff
@@ -154,27 +156,57 @@ export class PostHogSurveys {
     }
 
     loadIfEnabled() {
-        const surveysGenerator = assignableWindow?.__PosthogExtensions__?.generateSurveys
+        const phExtensions = assignableWindow?.__PosthogExtensions__
 
-        if (!this.instance.config.disable_surveys && this._decideServerResponse && !surveysGenerator) {
-            if (this._surveyEventReceiver == null) {
-                this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-            }
+        if (!phExtensions) {
+            logger.error('PostHog Extensions not found.')
+            return
+        }
 
-            assignableWindow.__PosthogExtensions__?.loadExternalDependency?.(this.instance, 'surveys', (err) => {
-                if (err) {
-                    return logger.error('Could not load surveys script', err)
-                }
+        const generateSurveys = phExtensions.generateSurveys
 
-                this._surveyManager = assignableWindow.__PosthogExtensions__?.generateSurveys?.(this.instance)
-            })
+        const disableSurveys = this.instance.config.disable_surveys
+
+        if (disableSurveys) {
+            logger.info('Disabled. Not loading surveys.')
+            return
+        }
+
+        if (!this._decideServerResponse) {
+            logger.warn('Decide not loaded yet. Not loading surveys.')
+            return
+        }
+
+        if (this._surveyEventReceiver == null) {
+            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
+        }
+
+        if (!generateSurveys) {
+            const loadExternalDependency = phExtensions.loadExternalDependency
+
+            if (loadExternalDependency) {
+                loadExternalDependency(this.instance, 'surveys', (err) => {
+                    if (err) {
+                        logger.error('Could not load surveys script', err)
+                        return
+                    }
+
+                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                })
+            } else {
+                logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
+            }
+        } else {
+            this._surveyManager = generateSurveys(this.instance)
```

Comment:
do we know `loadIfEnabled` is only called once?

we can maybe short circuit too if `this._surveyManager` exists but I haven't thought through what all this is doing :)

Reply:
Good point, didn't think of that since the goal was just more logging, but the surveys bit has room for lots of improvements.

Reply:
can be called by `onRemoteConfig`, `set_config` and `init`.
`set_config` can be called in many cases, so yeah it makes sense to short circuit.

Reply:
done, thanks.